libraries/AP_OSD/AP_OSD.h

```diff
@@ -380,7 +380,7 @@ class AP_OSD_ParamScreen : public AP_OSD_AbstractScreen
     static const uint8_t NUM_PARAMS = 9;
     static const uint8_t SAVE_PARAM = NUM_PARAMS + 1;
 
-#if HAL_WITH_OSD_BITMAP || HAL_WITH_MSP_DISPLAYPORT
+#if OSD_ENABLED && (HAL_WITH_OSD_BITMAP || HAL_WITH_MSP_DISPLAYPORT)
```

Comment:
Why do we need this ?

Reply:
Since, OSD_ENABLED = 0 compiles out [this whole](https://github.com/ArduPilot/ardupilot/blob/75b9a3ff7767e0bc78c6e8ca8f5b9b983c7c1b15/libraries/AP_OSD/AP_OSD_ParamScreen.cpp#L201-L609) code, the definition of `AP_OSD_ParamScreen::draw(void)` is also compiled out. And we override this function and don't provide the definition for the same, we get _symbol(s) not found_ error while linking the binaries. That's why if decided to compile this line out. Since, I just have started contributing to ardupilot, I am not aware of few things. I just went on by eliminating the code which was causing errors. I'll do it some other way if you want.